v8/payloads/interactions.ts

```diff
@@ -115,17 +115,24 @@ export interface APIApplicationCommandInteractionData {
 /**
  * https://discord.com/developers/docs/interactions/slash-commands#interaction-applicationcommandinteractiondataoption
  */
-export interface APIApplicationCommandInteractionDataOption {
-	name: string;
-	/**
-	 * The value returned here depends on the `ApplicationCommandOptionType` type of the option with the name
-	 * that matches this interface's `name`.
-	 *
-	 * You will need to manually cast this to the appropriate type based on the returned data
-	 */
-	value?: unknown;
-	options?: APIApplicationCommandInteractionDataOption[];
-}
+export type APIApplicationCommandInteractionDataOption =
+	| { name: string; options: APIApplicationCommandInteractionDataOption[] }
```

Comment:
Please split all these like the Command options ones :3

Reply:
This wasn't resolved yet it was marked as such 👀

Reply:
That line [wasn't supposed to be there](https://github.com/discordjs/discord-api-types/pull/72#discussion_r575765745) and was removed in 9318e8d :thinking: What did you want me to do with it?

Reply:
> Please split all these like the Command options ones :3

I think this is clear enough... Take all possibilities for the DataOption and make them interfaces, just like the CommandOption ones

Reply:
Ohh sorry I misunderstood what you were asking, I'll do that.